download-markdown/download-markdown.js

```diff
@@ -0,0 +1,172 @@
+      console.warn('Markdown not found at url', e.message);
+    }
+  }
+
+  return '';
+}
+
+/**
+ * Downloads a markdown file from a given document URL and saves it to a specified folder.
+ *
+ * @param {string} documentUrl - The URL of the markdown document to download.
+ * @param {string} folderPath - Folder where the downloaded markdown file will be saved.
+ * @param {string} entry - The name of the downloaded markdown file (without the file extension).
+ * @param {Function} [fetchFn=fetch] - The fetch function to use for making HTTP requests.
+ * @returns {Promise<boolean>} - true if the download is successful, or false otherwise.
+ */
+export async function downloadMD(documentUrl, folderPath, entry, fetchFn = fetch) {
+  if (ALLOW_SKIP && fs.existsSync(path.join(folderPath, `${entry}.md`))) {
+    console.log(`Skipping ${entry}.md`);
+    return true;
+  }
+
+  const waitMs = documentUrl.includes('hlx.page') ? PAGE_DELAY : LIVE_DELAY;
+  const markdown = await fetchMarkdown(`${documentUrl}.md`, waitMs, fetchFn);
+  const markdownFile = path.join(folderPath, `${entry}.md`);
+
+  if (!markdown) {
+    return false;
+  }
+
+  console.log(`Saving ${markdownFile}`);
+  saveToFile(markdownFile, markdown);
+
+  return true;
+}
+
+/**
+ * Downloads multiple markdown files from the specified URLs and saves them to a specified folder.
+ *
+ * @param {Map<string, string>} stagedUrls - A map of entry names to their corresponding URLs.
+ * @param {string} folderPath - The path of the folder where the markdown files will be saved.
+ * @param {Function} [fetchFn=fetch] - The fetch function to use for downloading the files.
+ * @returns {Promise<string[]>} - A list of entries that failed to download.
+ */
+export async function downloadMDs(stagedUrls, folderPath, fetchFn = fetch) {
+  const failedEntries = [];
+  for (const [entry, stageUrl] of stagedUrls) {
+    const success = await downloadMD(stageUrl, folderPath, entry, fetchFn);
+
+    if (!success) {
+      console.warn(`No markdown found for ${entry}`);
+      failedEntries.push(entry);
+    }
+  }
+  return failedEntries;
+}
+
+/**
+ * Downloads markdown files from a list of URLs and saves them to a specified folder.
+ *
+ * @param {string} folder - The folder path where the markdown files will be saved.
+ * @param {Array} list - The list of entries to be downloaded.
+ * @param {Array} locales - The locales to be used for localizing the staged URLs.
+ * @param {string} siteURL - The base URL of the website.
+ * @param {string} stagePath - The path to the staging environment.
+ * @returns {Promise<void>} A promise that resolves when the download process is complete.
+ */
+export function downloadMarkdown(folder, list, locales, siteURL, stagePath, fetchFn = fetch) {
+  // eslint-disable-next-line arrow-body-style
+  const stagedUrls = list.map((entry) => {
+    const entryPath = entryToPath(entry);
+    return [entryPath, localizeStageUrl(siteURL, entryPath, stagePath, locales)];
+  });
```

Comment:
```suggestion
  const stagedUrls = list.map((entry) => [entryToPath(entry), localizeStageUrl(siteURL, entryPath, stagePath, locales)]);
```
I imagine this may set off the linter via length though?